homeassistant/components/humidifier/__init__.py

```diff
@@ -171,7 +190,7 @@ def capability_attributes(self) -> dict[str, Any]:
 
         return data
 
-    @property
+    @cached_property
     def device_class(self) -> HumidifierDeviceClass | None:
         """Return the class of this entity."""
         if hasattr(self, "_attr_device_class"):
```

Comment:
Will the cache be invalidated, if `_attr_device_class` is not defined, but `entity_description.device_class` is?
I think it is possible that the `entity_description` is not provided at the time of the object creation, but calculated at a later stage (e.g. in `async_added_to_hass`)

Reply:
If `_attr_*` is changed, the setter will clear the cache.

The entity_description needs to be frozen and set when the object is created.  https://developers.home-assistant.io/blog/2023/12/11/entity-description-changes/

Reply:
The link only says it cannot be changed. But it is currently possible to define it late, and it would be accepted

Reply:
It's possible to do it but it wouldn't work as the base entity class also won't play well with it.

Reply:
Are you are aware of this actually happening or is this a theoretically discussion?

Reply:
I've been doing it in my custom integration. I will rewrite my own code, there is no issue here, I just thought it is something that is actually a supported feature an thus this change may break something. If not, then I am also happy.

Reply:
We would need to change the design of the entity base class to support changing out the entity description at runtime. 

I don't think that's something we want to support. The intent of freezing them is to prevent mistakes and complexity from entity descriptions changing at runtime.

Reply:
Ok, looks like adding entity description post entity creation is out of scope of this PR